facilitator/src/gcp_oauth.rs

```diff
@@ -140,6 +140,19 @@ impl OauthTokenProvider {
         })
     }
 
+    pub(crate) fn new_from_token(token: &str) -> OauthTokenProvider {
```

Comment:
This doesn't seem right -- `OauthTokenProvider` is responsible for obtaining the default account token from the GKE metadata service in `ensure_default_account_token`.

Reply:
I wrote this to handle the test cases that were just passing empty tokens. I'm not entirely sure what we want to do with those. E.g. in the case here: https://github.com/abetterinternet/prio-server/blob/7661a8f6eb5e581876fb01ee4227479e292c9154/facilitator/src/transport/gcs.rs#L164

Reply:
Any advice what would work well here?

Reply:
Oh, yeah, I see why you'd need to do that. Off the top of my head, you could either:

1. refactor `StreamingTransferWriter` so that it takes an `OauthTokenProvider`, or rather a trait object of some new trait that the existing `OauthTokenProvider` would implement. Then in the tests, you would provide an instance of a dummy `OauthTokenProvider` that just returns a token parsed from a fixed string.
2. provide variants on whatever function you add in `http.rs` to handle constructing that can take either `OauthTokenProvider` or `&str`, and `StreamingTransferWriter` would use the latter variant.

(1) would be kinda tricky, the borrow checker will yell at you a lot about `Box` and `dyn`, and refactoring how the transport stuff works is probably beyond the scope of this task.